src/query/api/v1/handler/prometheus/native/list_tags_test.go

```diff
@@ -0,0 +1,154 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package native
+
+import (
+	"encoding/json"
+	"math"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3/src/query/util/logging"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+type listTagsMatcher struct{}
+
+func (m *listTagsMatcher) String() string { return "list tags query" }
+func (m *listTagsMatcher) Matches(x interface{}) bool {
+	q, ok := x.(*storage.CompleteTagsQuery)
+	if !ok {
+		return false
+	}
+
+	if !q.Start.Equal(time.Time{}) {
+		return false
+	}
+
+	// NB: end time for the query should be roughly `Now`
+	diff := q.End.Sub(time.Now())
+	absDiff := time.Duration(math.Abs(float64(diff)))
+	if absDiff > time.Second {
+		return false
+	}
+
+	if !q.CompleteNameOnly {
+		return false
+	}
+
+	if len(q.FilterNameTags) != 0 {
+		return false
+	}
+
+	if len(q.TagMatchers) != 1 {
+		return false
+	}
+
+	return models.MatchAll == q.TagMatchers[0].Type
+}
+
+var _ gomock.Matcher = &listTagsMatcher{}
+
+func b(s string) []byte { return []byte(s) }
+func bs(ss ...string) [][]byte {
+	bb := make([][]byte, len(ss))
+	for i, s := range ss {
+		bb[i] = b(s)
+	}
+
+	return bb
+}
+
+func setupStorage(ctrl *gomock.Controller) storage.Storage {
+	store := storage.NewMockStorage(ctrl)
+	result := &storage.CompleteTagsResult{
+		CompleteNameOnly: true,
+		CompletedTags: []storage.CompletedTag{
+			{Name: b("bar")},
+			{Name: b("baz")},
+			{Name: b("foo")},
+		},
+	}
+
+	store.EXPECT().CompleteTags(gomock.Any(), &listTagsMatcher{}, gomock.Any()).
+		Return(result, nil)
+
+	return store
+}
+
+type writer struct {
+	results []string
+}
+
+var _ http.ResponseWriter = &writer{}
+
+func (w *writer) WriteHeader(_ int)   {}
+func (w *writer) Header() http.Header { return make(http.Header) }
+func (w *writer) Write(b []byte) (int, error) {
+	if w.results == nil {
+		w.results = make([]string, 0, 10)
+	}
+
+	w.results = append(w.results, string(b))
+	return len(b), nil
+}
+
+type result struct {
+	Status string   `json:"status"`
+	Data   []string `json:"data"`
+}
+
+func TestFind(t *testing.T) {
+	logging.InitWithCores(nil)
```

Comment:
been trying to push the standard stuff we have for zap now, could you use `src/x/test/NewLogger` to create the test logger and pass down to the appropriate place

Reply:
Called out in #1590 as well; all our other handlers are doing this at the moment, may be nicer to fix them all at once?

Reply:
I think we're currently using zap for logging; although we may do it in a bit of a weird way with a global zap logger

Reply:
Added this to #1590 under "General":
> Remove use of global Zap logger usage in favor of `src/x/instrument` logger which uses Zap and use `src/x/test/NewLogger` in tests